src/main/java/com/iexec/worker/sgx/SgxService.java

```diff
@@ -44,21 +56,40 @@ public SgxService(
     ) {
         this.workerConfigService = workerConfigService;
         this.dockerService = dockerService;
-        this.sgxEnabled = !SgxDriverMode.NONE.equals(sgxDriverMode) && isSgxSupported();
+        this.sgxDriverMode = sgxDriverMode;
+    }
+
+    @PostConstruct
+    void init() {
+        boolean sgxSupported = isSgxSupported(sgxDriverMode);
+        final boolean driverModeNotNone = SgxDriverMode.isDriverModeNotNone(sgxDriverMode);
+        if (!sgxSupported && driverModeNotNone) {
+            this.sgxEnabled = false;
+            log.error("SGX required but not supported by worker. Shutting down. " +
+                    "[sgxDriverMode: {}]", sgxDriverMode);
+            SpringApplication.exit(context, () -> 1);
```

Comment:
:+1:
Do we require the `return` statement below after calling this `exit` method ?

Reply:
That's not really mandatory, but I prefer to return as soon as possible, Otherwise, if code is modified later such as the following, there could be some side-effects.
```java
    @PostConstruct
    void init() {
        if (...) {
            ...
            SpringApplication.exit(context, () -> 1);
        }
        ...
        doSomethingWithSideEffects();
    }
```